tasks.c

```diff
@@ -211,7 +211,7 @@
 
     #define taskSELECT_HIGHEST_PRIORITY_TASK()                                                  \
     do {                                                                                        \
-        UBaseType_t uxTopPriority;                                                              \
+        UBaseType_t uxTopPriority = 0;                                                          \
```

Comment:
What do you think about moving this initialization into `portGET_HIGHEST_PRIORITY()`?  The initialization appears to be necessary only for some implementations of `portGET_HIGHEST_PRIORITY()`.  I am guessing in this case it is necessary because the x86-x64 implementations pass `&uxTopPriority` as a parameter to a function, and the compiler generates a warning without the initialization.

My thinking here is to avoid unnecessary CPU cycles in the task-switching path for most platforms.

Reply:
Nothing against. 
This was required because when running the x64 build the upper bytes weren't set to 0 (ended up with something like `0xcccc0005` for example) which caused an exception for trying access wrong memory address.

Reply:
Done!